samcli/commands/local/cli_common/invoke_context.py

```diff
@@ -345,6 +348,19 @@ def _clean_running_containers_and_related_resources(self) -> None:
         cast(WarmLambdaRuntime, self.lambda_runtime).clean_running_containers_and_related_resources()
         cast(RefreshableSamFunctionProvider, self._function_provider).stop_observer()
 
+    def _add_account_id_to_global(self, parameters) -> Dict|None:
+        # Attempt to get the Account ID from the current session (if any)
+        sts = boto3.client("sts")
+        try:
+            account_id = sts.get_caller_identity()["Account"]            
```

Comment:
Let's do
```suggestion
            account_id = sts.get_caller_identity().get("Account")         
```
instead to avoid throwing a `KeyError` here in the event that we get a bad response.